x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/PlannerUtils.java

```diff
@@ -232,11 +232,12 @@ public static ElementType toElementType(DataType dataType) {
         if (dataType == EsQueryExec.DOC_DATA_TYPE) {
             return ElementType.DOC;
         }
+        // TODO: Spatial types can be read from source into BYTES_REF, or read from doc-values into LONG
```

Comment:
As above, this is another place wherre we seem to imply that one field type can only generate one Element type, is this a blocker to return different data structures depnding on how the data is going to be consumed?

Reply:
Actually, this case is an easier one. We're planning to make different data types for the doc-values version, so there will be no ambiguity in this case. The planner will plan the type it needs, either the source type or the doc values type, and then methods like this will simply see these as two different types. This is at least the idea I've discussed with Nik, and I think it makes sense.

Reply:
Indeed, this is another place where we were assuming all DataTypes go to a single element type. I don't know the right way to handle this. Some kind of information in `dataType` or something. I dunno.

I'm quite ok waiting on solving this for another PR though. We *know* this is going to be a big part of the next PR.